api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java

```diff
@@ -44,561 +40,653 @@
+    }
+    return definition;
+  }
+
+  /**
+   * Helper method that constructs a {@link CohortReview} with the specified ids and count.
+   *
+   * @param cohortId
+   * @param cdrVersionId
+   * @param cohortCount
+   * @return
+   */
+  private CohortReview createNewCohortReview(Long cohortId, Long cdrVersionId, long cohortCount) {
+    CohortReview cohortReview = new CohortReview();
+    cohortReview.setCohortId(cohortId);
+    cohortReview.setCdrVersionId(cdrVersionId);
+    cohortReview.matchedParticipantCount(cohortCount);
+    cohortReview.setCreationTime(new Timestamp(System.currentTimeMillis()));
+    cohortReview.reviewedCount(0L);
+    cohortReview.reviewStatus(ReviewStatus.NONE);
+    return cohortReview;
+  }
+
+  /**
+   * Helper method that will populate all gender, race and ethnicity per the spcecified list of
+   * {@link ParticipantCohortStatus}.
+   *
+   * @param participantCohortStatuses
+   */
+  private void lookupGenderRaceEthnicityValues(List<ParticipantCohortStatus> participantCohortStatuses) {
+    Map<String, Map<Long, String>> concepts = genderRaceEthnicityConceptProvider.get().getConcepts();
+    participantCohortStatuses.forEach(pcs -> {
+      pcs.setRace(concepts.get(GenderRaceEthnicityType.RACE.name()).get(pcs.getRaceConceptId()));
+      pcs.setGender(concepts.get(GenderRaceEthnicityType.GENDER.name()).get(pcs.getGenderConceptId()));
+      pcs.setEthnicity(concepts.get(GenderRaceEthnicityType.ETHNICITY.name()).get(pcs.getEthnicityConceptId()));
+    });
+  }
+
+  private PageRequest createPageRequest(PageFilterRequest request) {
+    String sortColumn = "";
+    if (request instanceof ParticipantCohortStatuses) {
+      sortColumn = Optional.ofNullable(((ParticipantCohortStatuses) request).getSortColumn())
+        .orElse(ParticipantCohortStatusColumns.PARTICIPANTID).toString();
+    } else if (request instanceof ReviewFilter) {
+      sortColumn = Optional.ofNullable(((ReviewFilter) request).getSortColumn())
+        .orElse(ReviewColumns.STARTDATE).toString();
+    }
+    int pageParam = Optional.ofNullable(request.getPage()).orElse(CohortReviewController.PAGE);
+    int pageSizeParam = Optional.ofNullable(request.getPageSize()).orElse(CohortReviewController.PAGE_SIZE);
+    SortOrder sortOrderParam = Optional.ofNullable(request.getSortOrder()).orElse(SortOrder.ASC);
+    return new PageRequest()
+      .page(pageParam)
+      .pageSize(pageSizeParam)
+      .sortOrder(sortOrderParam)
+      .sortColumn(sortColumn);
+  }
+
+  /**
+   * Helper method to convert a collection of {@link FieldValue} to {@link ParticipantData}.
+   *
+   * @param rm
+   * @param row
+   * @param domain
+   */
+  private ParticipantData convertRowToParticipantData(Map<String, Integer> rm,
+                                                      List<FieldValue> row,
+                                                      DomainType domain) {
+    if (domain.equals(DomainType.DRUG)) {
+      return new Drug()
```

Comment:
What are the benefits to strongly typing all of these objects as drug, condition, procedure, etc.?

Seems like we could just return a single object type with a "type" field on it telling you whether you were dealing with a drug, condition, etc., and a superset of all the fields for these types... and this code and the UI code would become much simpler and easier to maintain.